File: app/src/main/java/com/example/user/biblejournal/model/database/NoteDao.java
package com.example.user.biblejournal.model.database;


import androidx.lifecycle.LiveData;
import androidx.room.Dao;
import androidx.room.Insert;
import androidx.room.Query;
import androidx.room.Update;

import java.util.List;

@Dao
public interface NoteDao {
    @Query("SELECT id, title, content, dateCreated, dateEdited, tags FROM NoteEntity WHERE state = 'ACTIVE'")
    LiveData<List<NoteEntity>> getAllNotes();

    @Insert
    void insertNote(NoteEntity noteEntity);

    @Update
    void updateNote(NoteEntity noteEntity);

    @Query("SELECT id, title, content, tags, dateCreated, dateEdited from NoteEntity where id = :id")
    NoteEntity getNoteById(int id);
}

File: app/src/main/java/com/example/user/biblejournal/writer/local/dao/NoteDao.kt
package com.example.user.biblejournal.writer.local.dao


import androidx.room.Dao
import androidx.room.Insert
import androidx.room.Query
import androidx.room.Update
import com.example.user.biblejournal.writer.local.entity.NoteEntity

@Dao
interface NoteDao {
    @get:Query("SELECT id, title, content, dateCreated, dateEdited, state, tags FROM NoteEntity WHERE state = 'ACTIVE'")
    val allNotes: List<NoteEntity>

    @Insert
    fun insertNote(noteEntity: NoteEntity)

    @Update
    fun updateNote(noteEntity: NoteEntity)

    @Query("SELECT id, title, content, dateCreated, dateEdited, state, tags from NoteEntity where id = :id")
    fun getNoteById(id: Long): NoteEntity
}
